### ISSUE TITLE
Add support for NEG type PRIVATE_SERVICE_CONNECT in NetworkEndpointGroup

### ISSUE THREAD

**Issue by danistrebel** [hashicorp/terraform-provider-google#11631]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

[Network endpoint groups](https://cloud.google.com/compute/docs/reference/rest/v1/networkEndpointGroups) can be used as an elegant LB backend for talking to a PSC service provider's service attachment.

It would be great to add the PRIVATE_SERVICE_CONNECT type to the NEG implementation for allowing for this use case.

### New or Affected Resource(s)

* [compute_network_endpoint_group](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_network_endpoint_group)

### Potential Terraform Configuration

```tf
# Propose what you think the configuration to take advantage of this feature should look like.
# We may not use it verbatim, but it's helpful in understanding your intent.

      - !ruby/object:Api::Type::Enum
        name: 'networkEndpointType'
        description: |
          Type of network endpoints in this network endpoint group.
          NON_GCP_PRIVATE_IP_PORT is used for hybrid connectivity network
          endpoint groups (see https://cloud.google.com/load-balancing/docs/hybrid).
          Note that NON_GCP_PRIVATE_IP_PORT can only be used with Backend Services
          that 1) have the following load balancing schemes: EXTERNAL, EXTERNAL_MANAGED,
          INTERNAL_MANAGED, and INTERNAL_SELF_MANAGED and 2) support the RATE or
          CONNECTION balancing modes.
        values:
          - :GCE_VM_IP_PORT
          - :NON_GCP_PRIVATE_IP_PORT
          - :PRIVATE_SERVICE_CONNECT # new
```

```tf
- !ruby/object:Api::Type::String
  name: 'pscTargetService'
  description: |
    The target service url used to set up private service connection to a Google API or a PSC Producer Service Attachment.
```

### References

* https://cloud.google.com/compute/docs/reference/rest/v1/networkEndpointGroups

**Comment by danistrebel** [comment]
The deployment works but unfortunately only for a network type of `default` and auto-subnet `default`.

As documented here: https://cloud.google.com/sdk/gcloud/reference/compute/network-endpoint-groups/create#--network

for PSC NEGs one should also be able to specify a `network` and `subnet` on the NEG resource.

**Comment by danistrebel** [comment]
The above mentioned issue is fixed in the Google provider >=4.31.0 by [this PR](https://github.com/hashicorp/terraform-provider-google/pull/12176)
